includes/class-wc-payments-order-service.php

```diff
@@ -180,7 +180,12 @@ public function update_order_status_from_intent( $order, $intent ) {
 				break;
 			case Intent_Status::REQUIRES_ACTION:
 			case Intent_Status::REQUIRES_PAYMENT_METHOD:
-				$this->mark_payment_started( $order, $intent_data );
+				if ( ! empty( $intent_data['error'] ) ) {
+					$this->unlock_order_payment( $order );
+					$this->mark_payment_failed( $order, $intent_data['intent_id'], $intent_data['intent_status'], $intent_data['charge_id'], $intent_data['error']['message'] );
```

Comment:
It would be great to just pass `$intent_data` for simplicity, but I see other places in the codebase will be not that quick to align with that, so I'm fine keeping it as is but wanted to flag it.